src/main/java/org/broadinstitute/hellbender/engine/spark/datasources/ReadsSparkSource.java

```diff
@@ -110,13 +113,26 @@ public ReadsSparkSource(JavaSparkContext ctx) {
      * @return the header for the bam.
      */
     public static SAMFileHeader getHeader(final JavaSparkContext ctx, final String filePath) {
-        final SAMFileHeader samFileHeader;
         try {
-             samFileHeader = SAMHeaderReader.readSAMHeaderFrom(new Path(filePath), ctx.hadoopConfiguration());
+            FileSystem fs = FileSystem.get(ctx.hadoopConfiguration());
+            Path path = fs.makeQualified(new Path(filePath));
+            if (fs.isDirectory(path)) {
+                FileStatus[] bamFiles = fs.listStatus(path, new PathFilter() {
+                    private static final long serialVersionUID = 1L;
+                    @Override
+                    public boolean accept(Path path) {
+                        return path.getName().startsWith("part-");
```

Comment:
Is this prefix alway guaranteed or is it configurable somewhere?  We should either use the variable where this is defined directly or extract it as a constant.  (I have no idea where it's defined, by maybe you do?)